hazelcast-client/src/main/java/com/hazelcast/client/connection/nio/ClientConnectionManagerImpl.java

```diff
@@ -78,9 +78,16 @@
 import static com.hazelcast.client.spi.properties.ClientProperty.HEARTBEAT_INTERVAL;
 import static com.hazelcast.client.spi.properties.ClientProperty.HEARTBEAT_TIMEOUT;
 
+/**
+ * Implementation of {@link ClientConnectionManager}.
+ */
 public class ClientConnectionManagerImpl implements ClientConnectionManager {
 
-    private final NonBlockingIOThreadOutOfMemoryHandler OUT_OF_MEMORY_HANDLER = new NonBlockingIOThreadOutOfMemoryHandler() {
+    protected final AtomicInteger connectionIdGen = new AtomicInteger();
+
+    protected volatile boolean alive;
+
+    private final NonBlockingIOThreadOutOfMemoryHandler outOfMemoryHandler = new NonBlockingIOThreadOutOfMemoryHandler() {
```

Comment:
Why didn't you make the field static since it doesn't need to be an instance field.